js/private/repo_toolchains.bzl

```diff
@@ -0,0 +1,36 @@
+"Helpers to locate toolchains from within repository rules"
+
+load("@aspect_bazel_lib//lib:repo_utils.bzl", "repo_utils")
+
+def yq_path(rctx):
+    """Find the path to the host resolved yq binary from a repository rule.
+
```

Comment:
nit: document here that `.exe` is append on windows if extension is not one of `.exe`, `.bat` and `.cmd`

Reply:
Documented below in return value.